pipeline/data_store.py

```diff
@@ -1422,19 +1424,25 @@ def save_and_commit( self, exists_ok=False, overwrite=True, no_archive=False,
             if self.sub_image is not None:
                 self.sub_image.new_image = self.image  # update with the now-merged image
                 self.sub_image = self.sub_image.merge_all(session)  # merges the upstream_images and downstream products
-                self.ref_image.id = self.sub_image.ref_image_id  # just to make sure the ref has an ID for merging
+                self.sub_image.ref_image.id = self.sub_image.ref_image_id  # just to make sure the ref has an ID for merging
                 self.detections = self.sub_image.sources
 
             if self.detections is not None:
                 if self.cutouts is not None:
+                    if self.measurements is not None:  # keep track of which cutouts goes to which measurements
+                        for m in self.measurements:
+                            m._cutouts_list_index = self.cutouts.index(m.cutouts)
                     for cutout in self.cutouts:
                         cutout.sources = self.detections
                     self.cutouts = Cutouts.merge_list(self.cutouts, session)
 
                 if self.measurements is not None:
                     for i, m in enumerate(self.measurements):
-                        self.measurements[i].cutouts = self.cutouts[i]  # use the new, merged cutouts
-                        self.measurements[i] = session.merge(m)
+                        # use the new, merged cutouts
+                        self.measurements[i].cutouts = self.measurements[i].find_cutouts_in_list(self.cutouts)
+                        self.measurements[i].associate_object(session)
+                        self.measurements[i] = session.merge(self.measurements[i])
+                        self.measurements[i].object.measurements.append(self.measurements[i])
```

Comment:
Isn't this redundant?  Inside Measurement.associate_object, it sets the object field of the measurement.  That's pointing at the same relation as the measurements field of the object; from Object:

```
    measurements = orm.relationship(
        Measurements,
        back_populates='object',
        cascade='all, delete-orphan',
        passive_deletes=True,
        lazy='selectin',
        doc='All Measurements related to the object, can include duplicates or bad measurements! '
    )

```

Reply:
Can't merge the measurement without first associating an object. But when we associate the measurement still doesn't have an ID so adding it to the object.measurement relationship doesn't stick.